path_roles.go

```diff
@@ -36,13 +54,24 @@ func (r *terraformRoleEntry) toResponseData() map[string]interface{} {
 	}
 	if r.Organization != "" {
 		respData["organization"] = r.Organization
+		r.CredentialType = organizationCredentialType
 	}
 	if r.TeamID != "" {
 		respData["team_id"] = r.TeamID
+		// Default to legacy team credential type
+		if r.CredentialType == "" {
+			r.CredentialType = teamLegacyCredentialType
```

Comment:
Would the condition on lines 62 and 63 need to be `r.CredentialType == "" || r.CredentialType == teamLegacyCredentialType`? It can be explicitly set to `team_legacy` and therefore not empty, is that right?

Edit: Actually, is setting `r.CredentialType` necessary here at all? Since this function is just returning the Response map for the fields when read, perhaps we shouldn't be re-setting `r.CredentialType` at all. Let me know if I' missing why this is needed.

Reply:
yep thats a great catch!